GoogleDataTransport/GDTCORLibrary/GDTCORUploadPackage.m

```diff
@@ -76,30 +113,30 @@ - (void)completeDelivery {
                    @"It's an API violation to call -completeDelivery twice.");
   }
   _isDelivered = YES;
+  [_expirationTimer invalidate];
   if (!_isHandled && _handler &&
       [_handler respondsToSelector:@selector(packageDelivered:successful:)]) {
-    [_expirationTimer invalidate];
     _isHandled = YES;
     [_handler packageDelivered:[self copy] successful:YES];
   }
   GDTCORLogDebug(@"Upload package delivered: %@", self);
 }
 
 - (void)retryDeliveryInTheFuture {
+  [_expirationTimer invalidate];
   if (!_isHandled && _handler &&
       [_handler respondsToSelector:@selector(packageDelivered:successful:)]) {
-    [_expirationTimer invalidate];
     _isHandled = YES;
     [_handler packageDelivered:[self copy] successful:NO];
   }
   GDTCORLogDebug(@"Upload package will retry in the future: %@", self);
 }
 
-- (void)checkIfPackageIsExpired:(NSTimer *)timer {
+- (void)checkIfPackageIsExpired {
   if ([[GDTCORClock snapshot] isAfter:_deliverByTime]) {
+    [_expirationTimer invalidate];
```

Comment:
Would be nice to have a follow up PR with tests for the changes

Reply:
A unit test for a memory leak is included in this PR. This change is associated with that.